### PR TITLE
feat: boost unboost and relative tests

### PR DESCRIPTION
**Pull request by lsqrl**

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -167,6 +172,28 @@ contract Vault is IVault, ReentrancyGuard, Ownable {
...
+        uint256 stakingCap = vaults[token].stakingCap;
+        if (totalWealth + amount > stakingCap) revert Vault__Staking_Cap_Exceeded(token, totalWealth, stakingCap);
+
+        (, amount) = IERC20(token).transferTokens(msg.sender, address(this), amount);
+
+        emit Boosted(msg.sender, token, amount);
+    }
+
+    function unboost(address token, uint256 amount) external override isValidAmount(amount) {
+        // todo: do we want to make an explicit revert if msg.sender does not have enough boosts?
```
Comment (OxMarco) [inline contracts/Vault.sol:190]: yes! assert that `boosters[msg.sender][token] >= amount`

File: test/integration/Lending/Vault.test.ts [hunk 2]
```diff
@@ -147,6 +148,36 @@ describe("Lending integration tests", function () {
       expect(validEvents?.length).equal(1);
     });
 
+    it("Vault: boost liquidity", async function () {
```
Comment (OxMarco) [inline test/integration/Lending/Vault.test.ts:151]: boost extra APY tests?

**Review by OxMarco — CHANGES_REQUESTED** [review]
To review
* check amount upon withdrawal
* test APY boost on other stakers

File: contracts/Vault.sol (hunk 1 above)
Comment (lsqrl) [reply contracts/Vault.sol:190]: ok: doing it and defining specific error message

File: test/integration/Lending/Vault.test.ts (hunk 2 above)
Comment (lsqrl) [reply test/integration/Lending/Vault.test.ts:151]: yes: they should be done

**Pull request merged**